pkg/operator/credential_request.go

```diff
@@ -59,10 +64,20 @@ func applyCredentialsRequest(client dynamic.Interface, recorder events.Recorder,
 		return nil, false, err
 	}
 
-	// Check only CredentialRequest.Generation.
-	// Currently, the object does not use annotations / labels, so we don't sync them.
-	if existing.GetGeneration() == expectedGeneration && !forceRollout {
-		return existing, false, err
+	// Check CredentialRequest.Generation.
+	needApply := forceRollout
+	if existing.GetGeneration() != expectedGeneration {
+		needApply = true
+	}
```

Comment:
Do we need to check the generation if we're checking the spec hash?

Reply:
Yes, we need both hash and generation. Spec hash checks for changes in the operator (new images, new bindata, ...). Generation checks for changes in the cluster (user changing the object).